src/Type.cpp

```diff
@@ -7,33 +7,19 @@ namespace Halide {
 using std::ostringstream;
 
 /** Return an integer which is the maximum value of this type. */
-int Type::imax() const {
+int64_t Type::imax() const {
     if (is_uint()) {
-        if (bits == 32) {
-            return 0xffffffff;
-        } else if (bits < 32) {
-            return (int) ((1 << bits) - 1);
+        if (bits == 64) {
+            return -1;
```

Comment:
You are correct -- the uses of it cast it to an uint64_t, which means you know what the type is at the use, which means there may as well be a umax instead. I'll change it.